partner_company_type/models/res_partner.py

```diff
@@ -10,3 +11,33 @@ class ResPartner(models.Model):
     partner_company_type_id = fields.Many2one(
         comodel_name="res.partner.company.type", string="Legal Form"
     )
+
+    @api.constrains(
+        "country_id",
+        "state_id",
+        "partner_company_type_id",
+    )
+    def _check_partner_company_type_country_state(self):
+        for rec in self:
+            if not rec.partner_company_type_id:
```

Comment:
```suggestion
            if not rec.partner_company_type_id or not rec.partner_company_type_id.country_ids or not rec.partner_company_type_id.state_ids:
```
with this change you can then remove the first check in the other two 'if' statements